pkg/errors/config.go

```diff
@@ -0,0 +1,41 @@
+package errors
+
+import (
+	"fmt"
+	"strings"
+)
+
+// ConfigErrFn is a function that can create a new configuration error. You pass it a
+// message specific to the error you found when you were validating configuration,
+// and it knows how to print out the actual configuration name and other helpful information.
+type ConfigErrFn func(string) error
```

Comment:
@marwan-at-work I wanted to but can't because the config package imports mode which needs all the config error stuff. Do you think it should be somewhere besides there?

Reply:
Sorry, I meant the `mode` package...which makes me realize `ConfigErrFn` makes no sense here, since it's not a Config package error, it's a `mode` error :) 

So that said, all of my suggestion are meant for keeping this functionality local to wherever the error is happening, in this case inside `mode.go` 

Thanks @arschles :v: